### PR TITLE
Add `fastlane-env-reminder` GitHub Action

### PR DESCRIPTION
**Pull request by mollyIV**
### Motivation and Context

> Some time ago GitHub has announced GitHub Actions. Recently a fastlane organisation got an access to Actions Beta. It is a good time to start playing with the new feature from GitHub.

It is the first step of bot's code migration to GitHub Actions based on [this proposal](https://github.com/fastlane/fastlane/issues/15078). 

This PR is introducing an action for adding a comment to include `fastlane env` in an issue description.

closes https://github.com/fastlane/github-actions/issues/1.
reference to the epic: https://github.com/fastlane/github-actions/issues/3.

### Description

I followed the instructions from [`Creating a JavaScript Action`](https://github.com/actions/toolkit/blob/master/docs/javascript-action.md) guidelines and [`Getting Started: GitHub context`](https://github.com/actions/toolkit/pull/80) pull request. 

I was looking at the examples from [GitHub Actions repository](https://github.com/actions), i.e. [first-interaction](https://github.com/actions/first-interaction).

Please note that this is just an action implementation. It **does not take** any effect itself. We still need to create the **[workflow](https://help.github.com/en/articles/configuring-a-workflow#creating-a-workflow-file)**, which will be using the `add-fastlane-env-comment` action. I will create dedicated issue for that once we merge this PR. 

There's a lot of changes in this PR because I needed to commit production dependencies. To make a review easier, I did split the commits:
* `Add GitHub Action for adding a comment to include `fastlane env` in an issue description` 👈 actual code changes
* `Add production dependencies` 👈 adding production dependencies

We proposed a solution **how to avoid such a big PR** [in the comments below](https://github.com/fastlane/github-actions/pull/2#issuecomment-528069932).

### Testing

To test the changes, I implemented the unit tests. To run them, execute `npm test` (followed by `npm install` to install the package dependencies and `npm build` to build the package). 

image

🙇 🙇

### CODE REVIEW

**Comment by janpio** [comment]
Is there no way to do the "commit bundled dependencies" step automatically in another branch so the main branch is clean? (Maybe there is a Github Action for that... ;) )

**Comment by mollyIV** [comment]
> Is there no way to do the "commit bundled dependencies" step automatically in another branch so the main branch is clean? (Maybe there is a Github Action for that... ;) )

That would be lovely @janpio ❤️  

It is hard to read the PR changes due to `node_modules` dependencies. To make it easier, I split the commits. I had a look at [GitHub Actions repository](https://github.com/actions?utf8=%E2%9C%93&q=&type=&language=javascript) and it looks for `javascript` projects they were also creating big PRs. 

An alternative was to zip the dependencies, like they did in [first-interaction](https://github.com/actions/first-interaction) action. However I can imagine that updating the dependencies will be harder than updating them in `package.json`.

**Comment by mollyIV** [comment]
👋 @janpio,

added `node_modules` to `.gitattributes` file, so the generated files are not rendered by default. I hope it will make a review a bit easier. 

image

**Comment by mollyIV** [comment]
I did read a [`Creating a JavaScript Action` documentation](https://github.com/actions/toolkit/blob/master/docs/javascript-action.md) section. Maybe we could leverage described `Dev Workflow`:

> The workflow below describes one possible workflow with a branching strategy. Others exist.

> Key Point: the branch that users reference in their workflow files should reference an action that has only the production dependencies.

> The workflow below describes a strategy where you code in master (with node_modules ignored) with a v1 branch users reference and contains the product references. Actions are self contained referenced on the github graph of repos.

Especially GitHub will provide tooling and an action to automate it. According to [`Publish a v1-release Action`](https://github.com/actions/toolkit/blob/master/docs/javascript-action.md#publish-a-v1-release-action):
> NOTE: We will provide tooling and an action to automate this soon.

So basically we would have `master` branch (with `node_modules` ignored, so easy to read PRs) and `release` branch (or something similar, to get a nice reference point in `yml` file - `-uses: fastlane/github-action/fastlane-env-reminder@release`).

We could also consider tagging the release, however we will have multiple actions in the same repo. 

What do you think? 🤔

**Comment by janpio** [comment]
Ja, that sounds _so_ much more sane.

Is this doable manually for now?

> We could also consider tagging the releases, however we will have multiple actions in the same repo so it might be an issue. Tagging the releases should be a go for a single, public action though.

You can always prefix the tag and release with the project name to be able to use this functionality properly. As we want to have multiple actions in here, we might have to do that anyway (unless we want to release them all at the same time).

**Comment by mollyIV** [comment]
> Is this doable manually for now?

Totally 👍We can add production dependencies on `latest` branch manually without any issues. 

> You can always prefix the tag and release with the project name to be able to use this functionality properly.

Sure 👍 Initially, the idea is to release all the actions at the same time. So we will be creating a PR to the `latest` branch with adding only production dependencies for all the actions. Obviously the flow is gonna be different for "public" actions.

File: fastlane-env-reminder/docs/contributors.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+### Build and Test
+
+```bash
+$ npm run build
+$ npm test
+```
+
+### Checkin
+
+- Do checkin source (src)
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
- Do checkin source (`src`)
```

File: fastlane-env-reminder/docs/contributors.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- Do checkin build output (lib)
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
- Do checkin build output (`lib`)
```

File: fastlane-env-reminder/docs/contributors.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+- Do not checkin runtime node_modules in development workflow (see [Dev Workflow documentation](https://github.com/actions/toolkit/blob/master/docs/javascript-action.md#dev-workflow))
+- Do not checkin devDependency node_modules (husky can help see below)
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
- Do not check in devDependency `node_modules` (husky can help, see below)
```

although this might not be necessary with the sentence before

File: fastlane-env-reminder/docs/contributors.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Contributors
+
+### Build and Test
+
+```bash
+$ npm run build
+$ npm test
+```
+
+### Checkin
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
### Check in
```

File: fastlane-env-reminder/docs/contributors.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+- Do checkin source (src)
+- Do checkin build output (lib)
+- Do not checkin runtime node_modules in development workflow (see [Dev Workflow documentation](https://github.com/actions/toolkit/blob/master/docs/javascript-action.md#dev-workflow))
+- Do not checkin devDependency node_modules (husky can help see below)
+
+### devDependencies
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
### `devDependencies`
```

File: fastlane-env-reminder/docs/contributors.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+In order to handle correctly checking in node_modules without devDependencies, we run [Husky](https://github.com/typicode/husky) before each commit.
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
In order to handle correctly checking in `node_modules` without `devDependencies`, we run [Husky](https://github.com/typicode/husky) before each commit.
```
is this still true?

File: fastlane-env-reminder/docs/contributors.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+This step ensures that formatting and checkin rules are followed and that devDependencies are excluded. To make sure Husky runs correctly, please use the following workflow:
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
This step ensures that formatting and checkin rules are followed and that `devDependencies` are excluded. To make sure Husky runs correctly, please use the following workflow:
```

File: fastlane-env-reminder/docs/contributors.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+```
+npm install                                 # installs all devDependencies including Husky
+git add abc.ext                             # Add the files you've changed. This should include files in src, lib, and node_modules (see above)
+git commit -m "Informative commit message"  # Commit. This will run Husky
+```
+
+During the commit step, Husky will take care of formatting all files with [Prettier](https://github.com/prettier/prettier) as well as pruning out devDependencies using `npm prune --production`.
```
Comment (janpio) [inline fastlane-env-reminder/docs/contributors.md]:
```suggestion
During the commit step, Husky will take care of formatting all files with [Prettier](https://github.com/prettier/prettier) as well as pruning out `devDependencies` using `npm prune --production`.
```

File: fastlane-env-reminder/package.json [hunk 9]
```diff
@@ -0,0 +1,31 @@
+{
+  "name": "fastlane-env-reminder-action",
+  "version": "0.0.0",
```
Comment (janpio) [inline fastlane-env-reminder/package.json]: correct?

Comment (mollyIV) [reply fastlane-env-reminder/package.json]: Yes, that is correct 👍 It points to a workspace version.
